src/azure-cli/azure/cli/command_modules/serviceconnector/action.py

```diff
@@ -213,3 +213,24 @@ def get_action(self, values, option_string):  # pylint: disable=no-self-use
 
         d['auth_type'] = 'servicePrincipalSecret'
         return d
+
+
+class AddOptOut(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        action = self.get_action(values)
+        namespace.opt_out_set = action
+
+    def get_action(self, values):  # pylint: disable=no-self-use
+        opt_out_set = set()
+        opt_out_options = {'public-network', 'config'}
```

Comment:
Can the `authentication` step be opted-out?

can private-endpoint and private-link be otped-out? how about just using `network` and `configuration`?

Reply:
for private-endpoint and private-link, our api does not have an opt-out option. I will look into authentication opt-out.

Reply:
`config` is so ambiguous, could it be `configinfo` to align with swagger to give user some hint?

Reply:
> Can the `authentication` step be opted-out?
> 
> can private-endpoint and private-link be otped-out? how about just using `network` and `configuration`?

vnetsolution is by default off.

Reply:
> Can the `authentication` step be opted-out?
> 
> can private-endpoint and private-link be otped-out? how about just using `network` and `configuration`?

I don't think authentication can be opted out in our creation workflow.

Reply:
then for network part, i think we should use `network`, as the opt-out option covers both public network and vnet

for config part, how about use `appsettings`? `configinfo` seems also ambiguous